definitions/marts/looker_studio_marts/ecf/ls_ecf_correct_cohort_checker.sqlx

```diff
@@ -0,0 +1,255 @@
+        AND
+        earliest_dec_suggestion_match = 0
+
+    For ECTs in the "wrong" cohort without declarations:
+      WHERE
+        eligible_for_funding = TRUE
+        AND
+        participant_role = 'ECT'
+        AND 
+        declaration_id IS NULL
+        AND
+        niot_partnership_chk = 0
+        AND
+        start_date_suggestion_match = 0
+
+  Dependencies:
+    dataform.ecf_golden_thread
+    dataform.ls_declarations_provider_names
+    dataform.partnerships_latest_cpd
+
+  Version History:
+    1.0 2024-06-05 Tony Page
+      Initial Build
+*/
+
+-- Gather ECF Golden Thread
+WITH golden_thread AS( 
+  SELECT
+     -- Manually assign the user_id for specific participant profile where the profile no longer exists in latest_cpd data and has no superceeded record to call upon.
+     CASE 
+      WHEN egt.participant_profile_id = '8cd0928c-ecad-482f-95cd-6a0ff3e840ad' THEN '79dee4de-a6dd-407c-ab1c-690fcb6d0056'
+      ELSE egt.user_id
+     END AS user_id
+    ,egt.participant_profile_id
+    ,CASE
+      WHEN egt.participant_type LIKE '%ECT' THEN 'ECT'
+      WHEN egt.participant_type LIKE '%Mentor' THEN 'Mentor'
+      ELSE NULL
+     END AS participant_role
+    ,egt.schedule_identifier
+    ,egt.induction_start_date AS induction_start_date
+    ,egt.cohort
+    ,egt.eligible_for_funding
+    ,egt.partnership_id
+    ,prt.relationship AS partnership_relationship
+    ,egt.lead_provider_name AS induction_lead_provider_name
+  FROM 
+    ${ref('ecf_golden_thread')} AS egt
+  LEFT JOIN
+    ${ref('partnerships_latest_cpd')} AS prt
+  ON
+    prt.id = egt.partnership_id
+),
+
+-- Collect all s/e/p/p declarations relating to ECF in the appropriate states.
+declarations AS (
+  SELECT
+     id AS declaration_id
+    ,DATE(declaration_date) AS declaration_date
+    ,declaration_type
+    ,state AS declaration_state
+    ,participant_profile_id
+    ,cpd_lead_provider_id AS lead_provider_id
+    ,cpd_lp_name AS lead_provider_name
+    -- Calculate earliest declaration date for the participant for funded declarations only.
+    ,MIN(CASE WHEN state IN ('paid', 'payable', 'eligible') THEN DATE(declaration_date) ELSE DATE(2050,12,31) END) OVER (PARTITION BY participant_profile_id) AS earliest_declaration_date
+  FROM
+    ${ref('ls_declarations_provider_names')}
+  WHERE
+    programme = 'ECF'
+    AND
+    state IN (
+       'paid'
+      ,'payable'
+      ,'eligible'
+      ,'submitted'
+    )
+),
+
+-- Join declarations onto deduped induction records & add on the grouping of suggested cohorts by induction start date and earliest declaration..
+combined_data AS (
+  SELECT 
+     pro.user_id
+    ,pro.participant_profile_id
+    ,pro.participant_role
+    ,pro.schedule_identifier AS current_schedule
+    ,dec.declaration_id
+    ,dec.declaration_date
+    ,dec.lead_provider_name
+    ,IF(pro.induction_lead_provider_name = 'No lead provider', NULL, pro.induction_lead_provider_name) AS induction_lead_provider_name

```

Comment:
If you use ecf_declarations, you won't need this statement just use the cpd_lead_provider_name field

Reply:
There will be rows without declarations and in those situations we still require the associated Lead Provider which is against the induction record. Due to this I don't believe I can rely entirely on the ecf_declarations table.